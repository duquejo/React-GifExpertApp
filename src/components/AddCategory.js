import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

  const [ inputValue, setInputValue] = useState(''); // ''

  const handleInputChange = ( e ) => {
    setInputValue( e.target.value ); // Setting Input event
  };

  const handleSubmit = ( e ) => {
    
    e.preventDefault();
    // Greater than 2 value

    if( inputValue.trim().length > 2 ) {
      // setCategories( [...categories, 'Fullmetal Alchemist'] );
      setCategories( cats => [ inputValue, ...cats ] ); // Callback with the last state
      setInputValue( '' );
    }
  };

  return (
    <form onSubmit={ handleSubmit }>
      <input 
        type="text"
        value={ inputValue }
        onChange={ handleInputChange }
        placeholder="Type something and press 'enter' for giphy search!"
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
};