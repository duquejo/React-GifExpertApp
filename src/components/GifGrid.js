
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from "../hooks/useFetchGifs";
import PropTypes from 'prop-types';

export const GifGrid = ( { category } ) => {

  const { data:images, loading } = useFetchGifs( category );

  return (
    <>
      <h3 className="animate__animated animate__fadeIn"> { category } </h3>
      { loading && <p>Loading...</p> }
      <div className="card-grid">
        {
          images.map( img => <GifGridItem key={ img.id } { ...img } /> ) // Sending all properties with Object Rest operator
        }
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
};