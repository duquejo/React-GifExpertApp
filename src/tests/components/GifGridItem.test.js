import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('<GifGridItem /> unit tests cases', () => {
  
  const title = 'A title';
  const url = 'https://127.0.0.1/something.gif';
  const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

  test('Should show the component successfully', () => {

    expect( wrapper ).toMatchSnapshot();
  });

  test('Should have a paragraph with the title', () => {

    const paragraph = wrapper.find('p');
    expect( paragraph.text().trim() ).toBe( title );
  });

  test('Should have an image url and alt text ', () => {

    const image = wrapper.find('img');
    expect( image.prop('src') ).toBe( url );
    expect( image.prop('alt') ).toBe( title );
  });
  
  test('Should have the animate__fadeIn class', () => {

    const divContainer = wrapper.find('div');
    expect( divContainer.prop('className') ).toContain( 'animate__fadeIn' );
  });
  
});