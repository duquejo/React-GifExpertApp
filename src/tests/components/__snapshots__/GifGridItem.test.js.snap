// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> unit tests cases Should show the component successfully 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://127.0.0.1/something.gif"
  />
  <p>
     
    A title
     
  </p>
</div>
`;
